controllers/api_keys.rb

```diff
@@ -12,6 +12,7 @@ class CalendarEvents::ApiKeysController < ApplicationController
   TODO: Instead we should allow a unique key to be created for each client.
 =end
   def index
+    raise Discourse::NotFound unless current_user
     key = find_or_create!
```

Comment:
Following the pattern used in Discourse use,
```
before_action :ensure_logged_in

```

Reply:
Here you go